src/main/java/com/google/api/codegen/transformer/csharp/CSharpGapicSurfaceSnippetsTransformer.java

```diff
@@ -0,0 +1,190 @@
+import com.google.api.codegen.transformer.MethodTransformerContext;
+import com.google.api.codegen.transformer.ModelToViewTransformer;
+import com.google.api.codegen.transformer.ModelTypeTable;
+import com.google.api.codegen.transformer.ParamWithSimpleDoc;
+import com.google.api.codegen.transformer.SurfaceNamer;
+import com.google.api.codegen.transformer.SurfaceTransformerContext;
+import com.google.api.codegen.util.csharp.CSharpTypeTable;
+import com.google.api.codegen.viewmodel.ApiMethodType;
+import com.google.api.codegen.viewmodel.SnippetsFileView;
+import com.google.api.codegen.viewmodel.StaticLangApiMethodView;
+import com.google.api.codegen.viewmodel.StaticLangXApiView;
+import com.google.api.codegen.viewmodel.StaticLangXCommonView;
+import com.google.api.codegen.viewmodel.StaticLangXSettingsView;
+import com.google.api.codegen.viewmodel.ViewModel;
+import com.google.api.tools.framework.model.Field;
+import com.google.api.tools.framework.model.Interface;
+import com.google.api.tools.framework.model.Method;
+import com.google.api.tools.framework.model.Model;
+import com.google.common.collect.ImmutableList;
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class CSharpGapicSurfaceSnippetsTransformer implements ModelToViewTransformer {
+
+  private static final String XAPI_TEMPLATE_FILENAME = "csharp/gapic_snippets.snip";
+
+  private final GapicCodePathMapper pathMapper;
+  private final ImportTypeTransformer importTypeTransformer;
+  private final ApiMethodTransformer apiMethodTransformer;
+
+  public CSharpGapicSurfaceSnippetsTransformer(GapicCodePathMapper pathMapper) {
+    this.pathMapper = pathMapper;
+    this.importTypeTransformer = new ImportTypeTransformer();
+    this.apiMethodTransformer = new ApiMethodTransformer();
+  }
+
+  @Override
+  public List<ViewModel> transform(Model model, ApiConfig apiConfig) {
+    List<ViewModel> surfaceDocs = new ArrayList<>();
+    SurfaceNamer namer = new CSharpSurfaceNamer(apiConfig.getPackageName());
+
+    for (Interface service : new InterfaceView().getElementIterable(model)) {
+      SurfaceTransformerContext context =
+          SurfaceTransformerContext.create(
+              service,
+              apiConfig,
+              createTypeTable(apiConfig.getPackageName()),
+              namer,
+              new CSharpFeatureConfig());
+      addCommonImports(context);
+      SnippetsFileView snippets = generateSnippets(context);
+      surfaceDocs.add(snippets);
+    }
+
+    return surfaceDocs;
+  }
+
+  @Override
+  public List<String> getTemplateFileNames() {
+    return Arrays.asList(XAPI_TEMPLATE_FILENAME);
+  }
+
+  private ModelTypeTable createTypeTable(String implicitPackageName) {
+    return new ModelTypeTable(
+        new CSharpTypeTable(implicitPackageName),
+        new CSharpModelTypeNameConverter(implicitPackageName));
+  }
+
+  private void addCommonImports(SurfaceTransformerContext context) {
```

Comment:
You could make a CSharpCommonTransformer (or CSharpSharedTransformer) class which does work that is the same between the transformers, and could include:

* This function (addCommonImports)
* The definitions of pagedMethodAdditionalParams and other additional params
* makeParam